src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
 
+        // exercise
         spyFulfiledProcessor();
-
         this.fulfilledProcessor.processFulfilledOrder(order);
+
+        // verify
+        Assert.assertNull(this.fulfilledOrderList.getNext());
         Assert.assertNotNull(this.failedOrderList.getNext());
     }
 
+    // test case: When running thread in the FulfilledProcessor without a LocalMember, the method
+    // processFulfilledOrder() must not change OrderState to Failed and must remain in Fulfilled
+    // list.
     @Test
     public void testRunProcessLocalComputeOrderWithoutLocalMember()
             throws FogbowManagerException, UnexpectedException, InterruptedException {
-        
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         this.fulfilledProcessor = Mockito.spy(new FulfilledProcessor(REMOTE_MEMBER_ID,
                 this.tunnelingService, this.sshConnectivity,
                 DefaultConfigurationConstants.FULFILLED_ORDERS_SLEEP_TIME));
 
+        // exercise
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
 
-        Thread.sleep(MIN_SLEEP_TIME);
-
+        // verify
         Assert.assertNotNull(this.fulfilledOrderList.getNext());
+        Assert.assertNull(this.failedOrderList.getNext());
     }
 
-    /**
-     * Test if a fulfilled order of an active localidentity compute instance has not being changed
-     * to failed if SSH connectivity is reachable.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is Ready, the
+    // method processFulfilledOrder() must not change OrderState to Failed and must remain in
+    // Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceReachable() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
         orderInstance.setState(InstanceState.READY);
         order.setInstanceId(FAKE_INSTANCE_ID);
 
+        // exercise
         mockCloudConnectorFactory(orderInstance);
-
         Mockito.when(this.sshConnectivity
                 .checkSSHConnectivity(Mockito.any(SshTunnelConnectionData.class))).thenReturn(true);
 
-        Assert.assertNull(this.failedOrderList.getNext());
-
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
+        this.thread.sleep(DEFAULT_SLEEP_TIME);
 
-        Thread.sleep(MIN_SLEEP_TIME);
+        // verify
+        Mockito.verify(this.sshConnectivity)
```

Comment:
Wouldn't it be better to specify the amount of times the method should be called?

Reply:
Fixed.